### PR TITLE
[Templates] Generate single test HTTPS Certificate

### PR DESCRIPTION
**Pull request by javiercn**
* Generates a single HTTPS certificate for tests instead of one per test.

### CODE REVIEW

**Comment by amcasey** [comment]
Could we just check one in?  I'm pretty sure the kestrel tests just have a folder of pre-made certs with various characteristics.

**Review by halter73 — APPROVED** [review]

File: src/ProjectTemplates/Shared/DevelopmentCertificate.cs [hunk 1]
```diff
@@ -1,42 +1,61 @@
...
+    public static DevelopmentCertificate Get(Assembly assembly)
     {
-        CertificatePath = certificatePath;
-        CertificatePassword = certificatePassword;
-        CertificateThumbprint = certificateThumbprint;
-    }
+        string[] locations = [
+            Path.Combine(AppContext.BaseDirectory, "aspnetcore-https.json"),
+            Path.Combine(Environment.CurrentDirectory, "aspnetcore-https.json"),
+            Path.Combine(Path.GetDirectoryName(assembly.Location)!, "aspnetcore-https.json")
```
Comment (halter73) [inline src/ProjectTemplates/Shared/DevelopmentCertificate.cs:20]: Can you add a comment explaining why the json file could end up in any of these three directories? It's only ever generated by PrepareForTest.targets, right?

File: src/ProjectTemplates/TestInfrastructure/PrepareForTest.targets [hunk 2]
```diff
@@ -22,6 +22,32 @@
...
   </ItemGroup>
 
+    <Target Name="GenerateTesDevCert" BeforeTargets="AssignTargetPaths">
+
+    <PropertyGroup>
+      <_DevCertFileName>aspnetcore-https.pfx</_DevCertFileName>
+      <_DevCertJsonFileName>aspnetcore-https.json</_DevCertJsonFileName>
+      <_DevCertPath>$(ArtifactsTmpDir)$(_DevCertFileName)</_DevCertPath>
+      <_DevCertJsonPath>$(ArtifactsTmpDir)$(_DevCertJsonFileName)</_DevCertJsonPath>
+      <_DotNetHostFileName Condition="$([MSBuild]::IsOSPlatform(`Windows`))">dotnet.exe</_DotNetHostFileName>
```
Comment (halter73) [inline src/ProjectTemplates/TestInfrastructure/PrepareForTest.targets:32]: Can this be removed?

```suggestion
```

**Review by MackinnonBuck — APPROVED** [review]

File: src/ProjectTemplates/Shared/DevelopmentCertificate.cs (hunk 1 above)
Comment (javiercn) [reply src/ProjectTemplates/Shared/DevelopmentCertificate.cs:20]: They are all the same 😄 

I was trying to figure out Helix as I couldn't find the cert (it wasn't being copied). I'll just use `appcontext.basedirectory`

Comment (javiercn) [inline src/ProjectTemplates/Shared/DevelopmentCertificate.cs:20]:
```suggestion
            Path.Combine(AppContext.BaseDirectory, "aspnetcore-https.json"),
```

**Pull request merged**